src/source/Dtls/Dtls.c

```diff
@@ -561,6 +569,33 @@ STATUS dtlsSessionPutApplicationData(PDtlsSession pDtlsSession, PBYTE pData, INT
     return retStatus;
 }
 
+STATUS dtlsSessionShutdown(PDtlsSession pDtlsSession)
+{
+    STATUS retStatus = STATUS_SUCCESS;
+    BOOL locked = FALSE;
+
+    CHK(pDtlsSession != NULL, STATUS_NULL_ARG);
+
+    MUTEX_LOCK(pDtlsSession->sslLock);
+    CHK(!ATOMIC_LOAD_BOOL(&pDtlsSession->shutdown), retStatus);
+    CHK(ATOMIC_LOAD_BOOL(&pDtlsSession->sslInitFinished), retStatus);
```

Comment:
Perhaps this one also needs to be moved above the lock

Reply:
same reason.